### PR TITLE
Add content source kind support for training sources

### PR DESCRIPTION
**Pull request by samvaity**

### CODE REVIEW

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/documentanalysis/implementation/util/Transforms.java [hunk 1]
```diff
@@ -785,12 +786,12 @@ public static Map<String, com.azure.ai.formrecognizer.documentanalysis.implement
         tags.forEach((key, classifierDocumentTypeDetails) -> {
             com.azure.ai.formrecognizer.documentanalysis.implementation.models.ClassifierDocumentTypeDetails innerClassifyDocTypeDetails
                 = new com.azure.ai.formrecognizer.documentanalysis.implementation.models.ClassifierDocumentTypeDetails();
-            if (classifierDocumentTypeDetails.getTrainingDataContentSource() instanceof BlobFileListContentSource) {
+            if (classifierDocumentTypeDetails.getTrainingDataContentSource().getKind().equals(TrainingDataContentSourceKind.AZURE_BLOB_FILE_LIST)) {
```
Comment (mssfang) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/documentanalysis/implementation/util/Transforms.java:789]:
```suggestion
            if (TrainingDataContentSourceKind.AZURE_BLOB_FILE_LIST.equals(classifierDocumentTypeDetails.getTrainingDataContentSource().getKind())) {
```

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/documentanalysis/implementation/util/Transforms.java [hunk 2]
```diff
@@ -785,12 +786,12 @@ public static Map<String, com.azure.ai.formrecognizer.documentanalysis.implement
...
                 innerClassifyDocTypeDetails.setAzureBlobFileListSource(
                     new com.azure.ai.formrecognizer.documentanalysis.implementation.models.AzureBlobFileListContentSource(
                         ((BlobFileListContentSource) classifierDocumentTypeDetails.getTrainingDataContentSource()).getContainerUrl(),
                         ((BlobFileListContentSource) classifierDocumentTypeDetails.getTrainingDataContentSource()).getFileList()));
-            } else {
+            } else if (classifierDocumentTypeDetails.getTrainingDataContentSource().getKind().equals(TrainingDataContentSourceKind.AZURE_BLOB)) {
```
Comment (mssfang) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/documentanalysis/implementation/util/Transforms.java:794]:
```suggestion
            } else if (TrainingDataContentSourceKind.AZURE_BLOB.equals(classifierDocumentTypeDetails.getTrainingDataContentSource().getKind())) {
```

File: sdk/formrecognizer/azure-ai-formrecognizer/src/test/java/com/azure/ai/formrecognizer/documentanalysis/administration/DocumentModelAdministrationClientTestBase.java [hunk 3]
```diff
@@ -96,7 +96,7 @@ private void setMatchers() {
     }
     static TokenCredential getCredentialByAuthority(String endpoint) {
         String authority = TestUtils.getAuthority(endpoint);
-        if (authority == AzureAuthorityHosts.AZURE_PUBLIC_CLOUD) {
+        if (authority.equals(AzureAuthorityHosts.AZURE_PUBLIC_CLOUD)) {
```
Comment (mssfang) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/test/java/com/azure/ai/formrecognizer/documentanalysis/administration/DocumentModelAdministrationClientTestBase.java:99]:
```suggestion
        if (AzureAuthorityHosts.AZURE_PUBLIC_CLOUD.equals(authority)) {
```

File: sdk/formrecognizer/azure-ai-formrecognizer/swagger/README.md [hunk 4]
```diff
@@ -63,7 +63,7 @@ directive:
 ### To run, use `autorest --tag:formrecognizer-documentanalysis README.md`
 ``` yaml $(tag) == 'formrecognizer-documentanalysis'
 use: '@autorest/java@4.1.13'
-input-file: https://raw.githubusercontent.com/Azure/azure-rest-api-specs/main/specification/cognitiveservices/data-plane/FormRecognizer/stable/2023-07-31/FormRecognizer.json
+input-file: ./FormRecognizer.json
```
Comment (mssfang) [inline sdk/formrecognizer/azure-ai-formrecognizer/swagger/README.md:66]: Change it to the actual URL Before merging it to the main.

Comment (samvaity) [reply sdk/formrecognizer/azure-ai-formrecognizer/swagger/README.md:66]: we need it withthis for some changes

**Review by mssfang — APPROVED** [review]

**Pull request merged**